client/src/app/components/dashboard/components/task-item/task-item.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { EditTask, Task } from '../../models/models.tasks';
+import { MatDialog } from '@angular/material/dialog';
+import { EditTaskComponent } from 'src/app/components/dialogs/edit-task/edit-task.component';
+import { DeleteTaskComponent } from 'src/app/components/dialogs/delete-task/delete-task.component';
+
+@Component({
+  selector: 'app-task-item',
+  templateUrl: './task-item.component.html',
+  styleUrls: ['./task-item.component.scss'],
+})
+export class TaskItemComponent implements OnInit {
+  @Input() completed: boolean = false;
+  @Input() taskItem!: Task;
+
+  @Output() completeTaskEvent = new EventEmitter<undefined>();
+  @Output() editTaskEvent = new EventEmitter<EditTask>();
+  @Output() deleteTaskEvent = new EventEmitter<void>();
+
+  checked = new FormControl(false);
+
+  constructor(public dialog: MatDialog) {
+    this.checked.valueChanges.subscribe((value) => {
+      if (value)
+        setTimeout(() => {
+          this.completeTaskEvent.emit();
+        }, 300);
```

Comment:
Why do we have this timeout?

Reply:
For better user experience; It waits for checkbox being checked before it removes the item upon completion.